manager/api_sdk_manager_test.go

```diff
@@ -0,0 +1,55 @@
+package manager
+
+import (
+	"net/http"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestGetSDKWithOauth2Authentication(t *testing.T) {
+
+	Convey("Given I have a request with a passthrough token in the header", t, func() {
+
+		req, _ := http.NewRequest("Get", "foo", nil)
+		req.Header.Add("Eric-Access-Token", `{"token_type":"bearer","access_token":"bar","expires_in":1234}`)
+
+		Convey("When I try to retrieve an instance of the SDK", func() {
+
+			service, err := GetSDK(req)
+
+			Convey("Then no errors should be returned", func() {
+
+				So(err, ShouldBeNil)
+
+				Convey("And the SDK service should not be nil", func() {
+
+					So(service, ShouldNotBeNil)
+				})
+			})
+		})
+	})
+}
+
+func TestGetSDKWithApiKeyAuthentication(t *testing.T) {
+
+	Convey("Given I have a request without a passthrough token in the header ", t, func() {
+
+		req, _ := http.NewRequest("Get", "foo", nil)
+
+		Convey("When I try to retrieve an instance of the SDK", func() {
+
+			service, err := GetSDK(req)
+
+			Convey("Then no errors should be returned", func() {
+
+				So(err, ShouldBeNil)
+
+				Convey("And the SDK service should not be nil", func() {
+
+					So(service, ShouldNotBeNil)
```

Comment:
From my ignorance: would there be an easy way to test that the returned service is the right one (API-key), not only it isn't null?

Reply:
The http client that is used to create the sdk is an unexported variable in the sdk. 
So I believe the only way would be to refactor the api-sdk to export this variable, which would then only be used for tests